src/libs/actions/Report.js

```diff
@@ -551,15 +551,19 @@ function updateReportActionMessage(reportID, sequenceNumber, message) {
             return;
         }
 
-        // If the message is deleted, update the last read in case the deleted message is being counted in the unreadActionCount
+        // If the message is deleted, we should
+        // 1. update the last read in case the deleted message is being counted in the unreadActionCount
+        // 2. Get the previous lastMessageText and update it in Onyx
         setLocalLastRead(reportID, lastReadSequenceNumbers[reportID]);
-    });
-
-    // If this is the most recent message, update the lastMessageText in the report object as well
-    if (sequenceNumber === reportMaxSequenceNumbers[reportID]) {
         Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, {
-            lastMessageText: ReportUtils.formatReportLastMessageText(message.text) || `[${Localize.translateLocal('common.deletedCommentMessage')}]`,
+            lastMessageText: ReportActions.getLastMessageText(reportID),
         });
+    });
+
+    // If this is the most recent message and it wasn't deleted, update the lastMessageText in the report object as well
+    const lastMessageText = ReportUtils.formatReportLastMessageText(message.text);
+    if (lastMessageText && sequenceNumber === reportMaxSequenceNumbers[reportID]) {
+        Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, {lastMessageText});
```

Comment:
I'm confused about why we are doing this step here. It looks like we would:

1. call this once

```js
Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, {lastMessageText})
``` 

2. the promise above resolves and then we set the `lastMessageText` again on line 559

Is that intentional?

Reply:
My memory is failing me and now I'm confused too 😅 I'll do some more testing and try to figure out why I thought this was necessary.